gpflow/training/monitor/monitor.py

```diff
@@ -0,0 +1,790 @@
+        be written to.
+        :param only_scalars: Restricts the list of output parameters to scalars.
+        :param parameters: List of model parameters to send to TensorBoard. If not
+        provided all parameters will be sent to TensorBoard.
+        :param additional_summaries: List of additional summary objects to send to TensorBoard.
+        """
+        super().__init__(model, event_path)
+        all_summaries = additional_summaries or []
+        parameters = parameters or list(model.parameters)
+
+        # Add scalar parameters
+        all_summaries += [tf.summary.scalar(p.full_name, tf.reshape(p.constrained_tensor, []))
+                          for p in parameters if p.size == 1]
+
+        # Add non-scalar parameters
+        if not only_scalars:
+            all_summaries += [tf.summary.histogram(p.full_name, p.constrained_tensor)
+                              for p in parameters if p.size > 1]
+
+        # Add likelihood
+        all_summaries.append(tf.summary.scalar("optimisation/likelihood",
+                                               model._likelihood_tensor))
+
+        # Create the summary tensor
+        self._summary = tf.summary.merge(all_summaries)
+
+
+class LmlTensorBoardTask(BaseTensorBoardTask):
+    """
+    Monitoring task that creates a TensorBoard with just one scalar value -
+    the unbiased estimator of the evidence lower bound (ELBO or LML).
+
+    The LML is averaged over a number of minimatches. The input dataset is split into the specified
+    number of sequential minibatches such that every datapoint is used exactly once. The set of
+    minibatches doesn't change from one iteration to another.
+
+    The task can display the progress of calculating LML at each iteration (how many minibatches
+    are left to compute). For that the `tqdm' progress bar should be installed (pip install tqdm).
+    """
+
+    def __init__(self, model: Model, event_path: str, minibatch_size: Optional[int] = 100,
+                 display_progress: Optional[bool] = True) -> None:
+        """
+        :param model: Model tensor
+        :param event_path: Path of the event file where the summary protocol buffers will
+        be written to.
+        :param minibatch_size: Number of points per minibatch
+        :param display_progress: if True the task displays the progress of calculating LML.
+        """
+
+        super().__init__(model, event_path)
+        self._minibatch_size = minibatch_size
+        self._full_lml = tf.placeholder(settings.tf_float, shape=())
+        self._summary = tf.summary.scalar(model.name + '/full_lml', self._full_lml)
+        self._display_progress = display_progress
+
+    def run(self, context: MonitorContext, *args, **kwargs) -> None:
+
+        with params_as_tensors_for(self._model):
+            tf_x, tf_y = self._model.X, self._model.Y
+
+        if self._display_progress:  # pragma: no cover
```

Comment:
Minor. For avoiding assigning 2x times wrapper with `lambda x: x` you can do this:
```
wrapper = None
if self._display_progress:
  ... pass ...
if wrapper is None:
   wrapper = lambda x: x
```

Reply:
Changed.